tests/Reduce/ToAverageTest.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools\Tests\Reduce;
+
+use IterTools\Reduce;
+use IterTools\Tests\Fixture\ArrayIteratorFixture;
+use IterTools\Tests\Fixture\GeneratorFixture;
+use IterTools\Tests\Fixture\IteratorAggregateFixture;
+
+class ToAverageTest extends \PHPUnit\Framework\TestCase
+{
+    protected const ROUND_PRECISION = 4;
+
+    /**
+     * @test         toAverage array
+     * @dataProvider dataProviderForArray
+     * @param        array $data
+     * @param        int|float $expected
+     */
+    public function testArray(array $data, $expected)
+    {
+        // Given: $data
+
+        // When
+        $result = Reduce::toAverage($data);
+
+        $this->roundIfFloat($expected, $result);
+
+        // Then
+        $this->assertSame($expected, $result);
```

Comment:
Suggestion: You can make use of the [assertEqualsWithDelta](https://phpunit.readthedocs.io/en/9.5/assertions.html#assertequalswithdelta) test method to simplify tests that have floating point values and not need to perform any arithmetic in between the method under test and the assertion, and thus, eliminating any extra code that might hide or introduce a bug in the code or tests.

Reply:
Done.